### ISSUE TITLE
Please respect 308 Permanent Redirect as Permanent

### ISSUE THREAD

**Issue by Flameeyes** [SSilence/selfoss#1178]
Context: https://www.ctrl.blog/entry/permanent-redirects.html

It seems like there's a few Selfoss users fetching one of my blog's _old_ feed addresses. And despite this responding with 308 Permanent Redirect for months, it seems like they don't learn the new address.

**Comment by jtojnar** [comment]
I knew it was only a matter of time before this issue is opened when I saw the post linked in some Reddit drama :smiley: 

It would be nice if we could just implement HTTP and RSS specifications properly. Unfortunately, many sites do not follow the HTTP specification correctly so we cannot trust that permanent redirect is actually meant to be permanent. This puts us in a weird place where we either get issues reported by owners of compliant sites or by users of uncompliant ones.

Another issue is that changing user entered data by a machine without user’s knowing is somewhat icky. If we allow that, a bug in selfoss or the site, or even a malicious actor in the middle can change it to an incorrect value permanently. (Notice the article you link even suggest checking the link over a month before changing it, which already deviates from the spec.)

For those reasons, for a long time, I had on my to-do list, implementing a UI that would inform user of encountered permanent redirects on the settings page of selfoss.

![#E79F3C](https://user-images.githubusercontent.com/705123/79231915-15c22e00-7e67-11ea-9d41-7954fd6bbecb.png)

Thinking about it more, we could even change it automatically if the permanent redirect remains over a period of time as the article suggests, as long we include an interactive item in the feed that informs user about the change and allows to easily revert it if necessary.

**Comment by jtojnar** [comment]
Possibly relevant: https://github.com/simplepie/simplepie/pull/660

**Comment by desbest** [comment]
What if someone does the blackhat SEO trick of buying expired domains in order to accumulate the high 4+ PageRank from the domain name? Or if a hacker temporarily hijacks the domain name?

Then the permenant redirect will take the Selfoss user to an unintended website, if the domain was to get put into its initial use, later on.

**Comment by jtojnar** [comment]
Yes that’s why I am leaning towards requiring explicit confirmation, see https://github.com/fossar/selfoss/issues/1178#issuecomment-613455621.
